src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets

```
@@ -41,6 +41,7 @@
     <_PackageReferenceExceptILCompiler Include="@(PackageReference)" Exclude="Microsoft.DotNet.ILCompiler" />
     <_ILCompilerPackageReference Include="@(PackageReference)" Exclude="@(_PackageReferenceExceptILCompiler)" />
     <KnownILCompilerPack Update="Microsoft.DotNet.ILCompiler" Condition="@(_ILCompilerPackageReference->'%(Identity)')=='Microsoft.DotNet.ILCompiler'">
+        <Warning Text="Adding an explicit package reference to 'Microsoft.DotNet.ILCompiler' works only on a narrow set of scenarios and can run into version errors if not properly used." />
```

```suggestion
        <Warning Text="Set PublishAot property to true and delete explicit `Microsoft.DotNet.ILCompiler` package reference in your project file. Explicit Microsoft.DotNet.ILCompiler package reference can run into version errors." />
```
I would make the warning text actionable. Also, I do not think there is a proper way to use the package reference that would avoid the versioning problems reliably.